projects/observability/src/shared/components/cartesian/d3/scale/cartesian-scale-builder.ts

```diff
@@ -186,19 +190,25 @@ export class CartesianScaleBuilder<TData> {
   }
 
   private get allSeries(): Series<TData>[] {
-    if (this.scaleState.series) {
-      return this.scaleState.series.filter(series => !series.hide);
-    }
+    const series = this.scaleState.series !== undefined ? this.scaleState.series.filter(s => !s.hide) : [];
+    const ranges =
+      this.scaleState.ranges !== undefined
+        ? this.scaleState.ranges.filter(r => !r.hide).flatMap(r => [r.upper, r.lower])
+        : [];
 
-    return [];
+    return [...series, ...ranges];
   }
 
   private get data(): TData[] {
-    if (this.scaleState.series) {
-      return this.scaleState.series.flatMap(series => series.data);
-    }
+    const series = this.scaleState.series !== undefined ? this.scaleState.series.flatMap(s => s.data) : [];
+    const ranges =
+      this.scaleState.ranges !== undefined
+        ? this.scaleState.ranges.flatMap(r =>
+            [r.upper.data.flatMap(d => d), r.lower.data.flatMap(d => d)].flatMap(d => d)
```

Comment:
This doesn't really make sense (flatmap shouldn't take an identity fn arg, and we shouldn't need 3 levels of flatmap for two levels of data). Maybe something like `this.scaleState.ranges.map(r => [r.lower.data, r.upper.data]).flat(2)`